src/cpp/utils/UnitsParser.cpp

```diff
@@ -77,7 +77,7 @@ uint32_t parse_value_and_units(
 
     to_uppercase(units);
 
-    std::regex pattern(R"(([kibmgKIBMG]{1,3})|(\d+))");
+    std::regex pattern(R"(([bB]{1})|([kibmgKIBMG]{2,3})|(\d+))");
```

Comment:
1. We already converted to uppercase, so we don't need the lowercase letters.
2. The 'at()` below is doing a search in the map (i.e. performing string comparisons) so, in a way, it is doing the job of the regex again.

This means we can remove the regex altogether, and just capture `out_of_range` around the `at()` operation, throwing the `invalid_argument` we have in line 83

An additional update we could make is adding the empty string to the map, with the same value we have for `"B"`, and remove the conditionals in the places where `parse_value_and_units` is being called

Reply:
I like this suggestion